backend/api/parse_pdf.py

```
@@ -1366,45 +1366,143 @@ async def extract_personas_and_key_figures_optimized(combined_content: str, titl
         debug_log(f"[ERROR] Persona extraction failed: {str(e)}")
         return _create_fallback_personas(title)
 
-async def generate_scenes_optimized(combined_content: str, title: str, session_id: str = None) -> list:
+async def generate_scenes_optimized(combined_content: str, title: str, key_figures: list = None, session_id: str = None) -> list:
```

Ruff (0.13.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)